langserver/references.go

```diff
@@ -166,6 +186,44 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSO
 	return locs, nil
 }
 
+// reverseImportGraph returns the reversed import graph for the workspace
+// under the RootPath. Computing the reverse import graph is IO intensive, as
+// such we may send down more than one import graph. The later a graph is
+// sent, the more accurate it is. The channel will be closed, and the last
+// graph sent is accurate. The reader does not have to read all the values.
+func (h *LangHandler) reverseImportGraph() <-chan importgraph.Graph {
+	// Ensure our buffer is big enough to prevent deadlock
+	c := make(chan importgraph.Graph, 1)
+
+	// Note: We use a background context since this operation should not
+	// be linked to an individual request.
+	ctx := context.Background()
+
+	go func() {
+		// TODO fetch from cache an inaccurate import graph
+
+		bctx := h.BuildContext(ctx)
+		h.importGraphOnce.Do(func() {
+			findPackageWithCtx := h.getFindPackageFunc()
+			findPackage := func(bctx *build.Context, importPath, fromDir string, mode build.ImportMode) (*build.Package, error) {
+				return findPackageWithCtx(ctx, bctx, importPath, fromDir, mode)
+			}
+			h.mu.Lock()
+			h.importGraph = tools.BuildReverseImportGraph(bctx, findPackage, h.FilePath(h.init.RootPath))
+			h.mu.Unlock()
+		})
+		h.mu.Lock()
+		// TODO(keegancsmith) h.importGraph may have been reset after once
```

Comment:
Part of my tasks this week is to make go-langserver much better at handling mutations. As part of that work I will be fixing up TODOs like this.